packedForest/src/forestTypes/unsupervisedForests/urerf/fpURerFBase.h

```diff
@@ -0,0 +1,148 @@
+#ifndef fpRerF_h
+#define fpRerf_h
+
+#include "../../../baseFunctions/fpForestBase.h"
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <unordered_map>
+#include <stdio.h>
+#include <ctime>
+#include <chrono>
+#include <cstdlib>
+#include "urerfTree.h"
+#include <sys/time.h>
+
+namespace fp {
+	template <typename T>
+		class fpURerFBase : public fpForestBase<T>
+	{
+		protected:
+			std::vector<urerfTree<T> > trees;
+			std::map<int, std::map<int, int> > simMat;
+                        std::map<std::pair<int, int>, double> pairMat;
+                        typedef Eigen::SparseMatrix<int> spMat;
+                        typedef Eigen::Triplet<int> TripType;
+                        std::vector<TripType> tripletList;
+                        SpMat eigenMat;
+		public:
+
+			~fpURerFBase(){}
+
+			fpDisplayProgress printProgress;
+			inline void printForestType(){
+				std::cout << "This is a urerf forest.\n";
+			}
+
+			inline void changeForestSize(){
+				trees.resize(fpSingleton::getSingleton().returnNumTrees());
+			}
+
+			inline void initSimMat(){
+				auto numObs = fpSingleton::getSingleton().returnNumObservations();
+				for(auto i = 0; i < numObs; ++i) {
+					std::map<int, int> init_map;
+					simMat[i] = init_map;
+				}
+			}
+
+                        inline void printSparseMat(){
```

Comment:
in principle, we don't want to maintain any code that isn't in use.  please delete any code which isn't needed.

Reply:
fixed in 483e171